ResearchKit/Common/ORKScaleSliderView.m

```diff
@@ -128,79 +159,68 @@ - (void)updateConstraints {
     
     NSDictionary *views = NSDictionaryOfVariableBindings(_slider,_leftRangeLabel,_rightRangeLabel, _valueLabel);
     
-    if ([_formatProvider isVertical])
-    {
+    if ([_formatProvider isVertical]) {
```

Comment:
The existing code was not great in this respect, and it's quite likely I missed it in a previous review, but it seems more likely that we should be setting up the constraints *once* at setup time rather than in updateConstraints. If you do it in updateConstraints you have to be quite careful to deal with old constraints, etc.

We know in advance whether we're vertical or not, and these constraints are totally fixed -- they're not dynamic.